code/modules/mob/living/simple_animal/friendly/mule_bot/mule_bot.dm

```diff
@@ -0,0 +1,202 @@
+
+/obj/effect/mule_link/Initialize()
+	. = ..()
+	var/mob/living/simple_animal/mule_bot/M = new(loc)
+	var/obj/item/remote/R = new(loc)
+	M.try_link(R)
+	R.bot = M
+	qdel(src)
```

Comment:
This seems like a dumb solution where you should be linking the remote with the mule instead

Reply:
added remote linking on attack using signals. also added signal to check if remote got deleted to automaticly unlink.

this still is convenient way though to spawn remote and bot from vendors

Reply:
Stinky if you ask me, but at least move it to the end of file